core/package/index.js

```diff
@@ -209,6 +209,9 @@ class Package {
     if (resolved.registry || resolved.type === "directory") {
       // a version (1.2.3) OR range (^1.2.3) OR directory (file:../foo-pkg)
       depCollection[depName] = `${savePrefix}${depVersion}`;
+      if (resolved.explicitWorkspace) {
+        depCollection[depName] = `workspace:${depCollection[depName]}`;
```

Comment:
As I mentioned in the newly-added test for this method, this isn't always the case (thinking specifically of prepublish resolution of `${protocol}:` versions)...